src/components/NewItem.js

```diff
@@ -0,0 +1,254 @@
+import React, { Component } from "react";
+import Axios from "axios";
+import Button from "@material-ui/core/Button";
+import ExpansionPanel from "@material-ui/core/ExpansionPanel";
+import ExpansionPanelSummary from "@material-ui/core/ExpansionPanelSummary";
+import ExpansionPanelDetails from "@material-ui/core/ExpansionPanelDetails";
+import Typography from "@material-ui/core/Typography";
+import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
+import TextField from "@material-ui/core/TextField";
+import Grid from "@material-ui/core/Grid";
+import Select from '@material-ui/core/Select';
+import MenuItem from '@material-ui/core/MenuItem';
+import FormControl from '@material-ui/core/FormControl';
+import InputLabel from '@material-ui/core/InputLabel';
+
+export default class NewItem extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      status: false,
+      updatedPrice: "",
+      updatedName: "",
+      updatedItem: "",
+      updatedImg: "",
+      units:""
+    };
+  }
+
+  updatePrice = (p) => {
+    let price = p.target.value;
+    this.setState({ updatedPrice: price });
+  };
+
+  updateImage = (pic) => {
+    let img = pic.target.value;
+    this.setState({ updatedImg: img });
+  };
+
+  updateName = (n) => {
+    let itemName = n.target.value;
+    this.setState({ updatedName: itemName });
+  };
+
+  updateUnits=(u)=>{
+    this.setState({units:u.target.value})
+  }
+
+  deleteItem = () => {
+    let id = { Id: this.props.item.Id };
+    Axios.post(
+      "https://murmuring-hamlet-58919.herokuapp.com/deleteItem",
+      id
+    ).then((res) => {      
+      if (res.status === 200 && res.data === "done") {        
+        this.props.delete(this.props.item.Id);
+      }
+    });
+  };
+
+  saveChanges = () => {
+    let currentList = this.props.item;
+    let newName = this.state.updatedName;
+    let newPrice = this.state.updatedPrice;
+    let newImg = this.state.updatedImg;
+    let newUnits = this.state.units
+    if (newName !== "") {
+      currentList.Name = newName;
+    }
+    if (newPrice !== "") {
+      currentList.Price = newPrice;
+    }
+    if (newImg !== "") {
+      currentList.Image = newImg;
+    }
+    if (newUnits !=="") {
+      currentList.Units = newUnits
+    }    
+    this.setState({ updatedItem: currentList });
+    let itemAndId = {
+      Id: this.props.item.Id,
+      Item: currentList,
+      Img: this.state.updatedImg,
+    };
+    Axios.post(
+      "https://murmuring-hamlet-58919.herokuapp.com/updateItem",
+      itemAndId
+    ).then((res) => {        
+        if (res.status === 200) {
+            return this.editItem()
+        }
+    });
+  };
+
+  editItem = () => {
+    if (!this.state.status) {
+      this.setState({ status: true });
+    } else {
+      this.setState({ status: false });
+    }
+  };
+
+  showMe = () => {
```

Comment:
move all this html to the render. you dont need a function if its all the html and you always call it